### ISSUE TITLE
Use ASM.JS as the target

### ISSUE THREAD

**Issue by drom** [stackgl/glsl-transpiler#31]
asm.js: http://asmjs.org/spec/latest/
Is supported chrome and firefox for quite some time.
Does it make sense generate it?

**Comment by dfcreative** [comment]
There is unfinished https://github.com/devongovett/glsl.js which attempted that.
Practically there are cases where we use _Array.prototype.map_ and other env methods. Not sure that goes well with asm.js.
Should understand that main performance drop is not because of suboptimal js, but due to absence of huge parallelism, like in GPU.

**Comment by drom** [comment]
@dfcreative agree, that the main performance drop is because of CPU is not GPU, but 2x average improvement maybe worth pursuing. Here is my comparison of asm / non-asm version of FFT:
https://github.com/drom/fourier/issues/1

**Comment by dfcreative** [comment]
Interesting! Would be nice to make comparative benchmark over [various implementations](https://github.com/scijs/fourier-transform/blob/master/benchmark.md)
As for glsl-transpiler, I am not sure there is anyone using it at all before making any serious changes. Because webgl is too well supported across all platforms.

**Comment by drom** [comment]
@dfcreative I see that my package is already in the list ;)

**Comment by dfcreative** [comment]
@drom your benchmarks are way more dependable and demonstrative :)

**Comment by drom** [comment]
@dfcreative I need to rerun it with node v6 / v7

**Comment by gre** [comment]
mmh but what about WebAssembly? 🙂

**Comment by dfcreative** [comment]
@gre I am watching for the WebAssembly area, do you have any easy getting-started resources by any chance? Seems like entry barrier is too expensive for ordinary js-developer

**Comment by gre** [comment]
yeah I don't have A LOT of experience in it frankly, but i've found these series of article quite interesting: https://hacks.mozilla.org/2017/02/a-cartoon-intro-to-webassembly/

**Comment by dfcreative** [comment]
@gre as far as I get this, wasm can be compiled from asm.js via [wasm](https://www.npmjs.com/package/wasm).

**Comment by jamen** [comment]
I've been learning and implement WASM the past week or so for [Elixir](https://elixir-lang.org/).  It has been quite fun!  My main resource so far has just been everything Web Assembly has put out.

 - [mbasso/awesome-wasm](https://github.com/mbasso/awesome-wasm) points to a bunch of great resources
 - [webassembly/spec](https://github.com/webassembly/spec) contains details you want to know not described in other places (especially for a compiler)
 - [webassembly/design](https://github.com/webassembly/design) gives a good abstract docs about wasm
 - [webassembly/binaryen](https://github.com/webassembly/binaryen) their toolchain.  Lets you compile, inspect, and debug outputs
 - [webassembly](https://github.com/webassembly) just their organization for a bunch of other neat projects

What I can say from my limited experience: The spec only describes [a few basic values](http://webassembly.github.io/spec/syntax/values.html), [some instructions](http://webassembly.github.io/spec/syntax/instructions.html) for working with them, and [a module format](http://webassembly.github.io/spec/syntax/modules.html) to define the program and reusability.

It does seem to be quite early though and still active in development (the binary format wasn't there when I first started investigating!) so it is probably worth watching the repos to see that discussion.  Note that they have several repos dedicated to sections of the spec like [webassembly/gc](https://github.com/webassembly/gc) for garbage collection, and so on.

Hope this helps and I hope to collab on some WASM stuff with people!

**Comment by dy** [comment]
Can be done via [lino](https://github.com/audio-lab/lino), which can be a better fit
